improve and test vite-plugin-cloudflare `.dev.vars` files support (#7864)
cloudflare/workers-sdk: packages/vite-plugin-cloudflare/README.md

  By default, the environment name is the Worker name with `-` characters replaced with `_`.
  Setting the name here will override this.

## Secrets

The vite plugin supports local development via secrets using the [`.dev.vars` file convention](https://developers.cloudflare.com/workers/configuration/secrets/#local-development-with-secrets).

Meaning that when you run `vite dev` or `vite preview` the plugin picks up secrets set in your `.dev.vars` files and makes them available to your application.

To instead make such secrets available to deployed applications follow the [relative Cloudflare documentation](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-on-deployed-workers).

> [!Warning]

> > Currently `vite preview` does not actually pick up `.dev.vars` files, the team is working on it and that should be available soon

Author: done 👍

> > Currently `vite preview` does not actually pick up `.dev.vars` files, the team is working on it and that should be available soon

Reviewer: My preference is that we don't mention `.dev.vars` in the docs until it is fully supported. I think having warnings can come across as a bit intense and put people off. There shouldn't be too much work to do once we've decided on what the best approach is for supporting them in preview. Let's have that discussion and then hopefully it can also be done in this PR.

> > Currently `vite preview` does not actually pick up `.dev.vars` files, the team is working on it and that should be available soon

Author: removed as the `vite preview` support has been added (d46955c41dff3d44ae4b26c1bc324bced2c43511)

> ## Secrets
> 
> The vite plugin supports local development via secrets using the [`.dev.vars` file convention](https://developers.cloudflare.com/workers/configuration/secrets/#local-development-with-secrets).
> 
> Meaning that when you run `vite dev` or `vite preview` the plugin picks up secrets set in your `.dev.vars` files and makes them available to your application.
> 
> > [!NOTE]
> > The `vite build` command copies the potential `.dev.vars` target file into the same output directory in which it creates the build output `wrangler.json`, this copied file is the one that `vite preview` will then use
> 
> To instead make such secrets available to deployed applications follow the [relative Cloudflare documentation](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-on-deployed-workers).
> 

Reviewer: I think this section can be replaced as suggested below and that it should go after the 'Cloudflare environments' section.

```suggestion
## Secrets

Secrets can be provided to your Worker in local development using the [`.dev.vars`](https://developers.cloudflare.com/workers/configuration/secrets/#local-development-with-secrets) file convention. If you are using [Cloudflare Environments](#cloudflare-environments) then the relevant `.dev.vars` file will be selected. For example, `CLOUDFLARE_ENV=staging vite dev` will load `.dev.vars.staging` if it exists and otherwise fall back to `.dev.vars`.

> [!NOTE]
> The `vite build` command copies the relevant `.dev.vars` file to the output directory. This is only used when running `vite preview` and is not deployed with your Worker.
```

> The default Vite environment name for a Worker is always the top-level Worker name.
> This enables you to reference the Worker consistently in your Vite config when using multiple Cloudflare environments.

> > [!NOTE]
> > Setting `CLOUDFLARE_ENV` also effects the `.dev.vars` being read, as files ending with a dot and the environment
> > name get prioritized over the standard `.dev.vars` ones (e.g. if I run `CLOUDFLARE_ENV=staging vite dev`, the vite plugin will
> > look for a `.dev.vars.staging` file to gather the secrets from, falling back to `.dev.vars` in case the file doesn't exist)
> 

Reviewer: This can be removed once the 'Secrets' section has been updated and moved.

```suggestion
```

If you run `vite build --mode staging` then the 'staging' Vite mode will be used and the 'staging' Cloudflare environment will be selected.
The value of `MY_VAR` will therefore be `'Staging var'`.

## Secrets

> Secrets can be provided to your Worker in local development using the [`.dev.vars`](https://developers.cloudflare.com/workers/configuration/secrets/#local-development-with-secrets) file convention. If you are using [Cloudflare Environments](#cloudflare-environments) then the relevant `.dev.vars` file will be selected. For example, `CLOUDFLARE_ENV=staging vite dev` will load `.dev.vars.staging` if it exists and otherwise fall back to `.dev.vars`.

Reviewer 2: ```suggestion
Secrets can be provided to your Worker in local development using a [`.dev.vars`](https://developers.cloudflare.com/workers/configuration/secrets/#local-development-with-secrets) file. If you are using [Cloudflare Environments](#cloudflare-environments) then the relevant `.dev.vars` file will be selected. For example, `CLOUDFLARE_ENV=staging vite dev` will load `.dev.vars.staging` if it exists and otherwise fall back to `.dev.vars`.
```

> [!NOTE]

> > The `vite build` command copies the relevant `.dev.vars` file to the output directory. This is only used when running `vite preview` and is not deployed with your Worker.

Reviewer 2: is this relevant?
```suggestion
> The `vite build` command copies the relevant `.dev.vars[.env-name]` file to the output directory. This is only used when running `vite preview` and is not deployed with your Worker.
```
